modules/drivers/bigquery-cloud-sdk/test/metabase/driver/bigquery_cloud_sdk_test.clj

```diff
@@ -240,6 +249,60 @@
                    {:source-table (mt/id view-name)
                     :order-by     [[:asc (mt/id view-name :id)]]}))))))))
 
+(deftest sync-table-with-required-filter-test
```

Comment:
I think you should add a test for materialized views too, based on this line 
https://github.com/metabase/metabase/pull/36677/files#diff-370f1464c7c10bd0d0bd84544e6f80102152617fcd99fea6e9bd4dbaff4903caR130

Reply:
right so turns out we currently can't figure out whether a materialized view have required a filter or not. See https://github.com/metabase/metabase/pull/36677/files#diff-370f1464c7c10bd0d0bd84544e6f80102152617fcd99fea6e9bd4dbaff4903caR131-R135 for comments.

I'll add this as a limitation in the PR

Reply:
Makes sense, thanks for investigating